webapp/src/components/systemConsole/questions/questions.tsx

```diff
@@ -0,0 +1,147 @@
+
+import type {SurveyQuestionsConfig} from 'types/mattermost-webapp';
+
+export type QuestionType = 'linear_scale' | 'text';
+
+const questionTypeDisplayName = new Map<QuestionType, string>([
+    ['linear_scale', 'Linear scale question (1 to 10)'],
+    ['text', 'Textual question'],
+]);
+
+export type Question = {
+    id: string;
+    title?: string;
+    text?: string;
+    type: QuestionType;
+    system: boolean;
+    mandatory: boolean;
+    helpText?: string;
+};
+
+function SurveyQuestions({id, setSaveNeeded, onChange, config, setInitialSetting}: CustomSettingChildComponentProp) {
+    const [questions, setQuestions] = useState<Question[]>([]);
+
+    const generateDefaultQuestions = (): Question[] => {
+        return [
+            {
+                id: utils.uuid(),
+                title: 'Survey message text',
+                text: 'Please take a few moments to help us improve your experience with Mattermost.',
+                type: 'text',
+                system: false,
+                mandatory: true,
+                helpText: 'This text will be sent in the bot message preceding the survey.',
+            },
+            {
+                id: utils.uuid(),
+                text: 'How likely are you to recommend Mattermost?',
+                type: 'linear_scale',
+                system: true,
+                mandatory: true,
+            },
+            {
+                id: utils.uuid(),
+                text: 'How can we make your experience better?',
+                type: 'text',
+                system: true,
+                mandatory: true,
+            },
+            {
+                id: utils.uuid(),
+                type: 'text',
+                system: false,
+                mandatory: false,
+            },
+        ];
+    };
+
+    useEffect(() => {
+        // Set initial value from saved config
+        const savedQuestions = config.PluginSettings.Plugins?.['com.mattermost.user-survey']?.systemconsolesetting.SurveyQuestions;
+
+        const initialSetting = savedQuestions || generateDefaultQuestions();
+        setQuestions(initialSetting);
+        setInitialSetting(id, initialSetting);
+    }, [config.PluginSettings.Plugins, id, setInitialSetting]);
+
+    const saveSettings = useCallback((setting: SurveyQuestionsConfig) => {
+        setSaveNeeded();
+        onChange(id, setting);
+    }, [id, onChange, setSaveNeeded]);
+
+    const questionOnChangeHandler = useDebouncedCallback(
+        useCallback((e: React.ChangeEvent<HTMLInputElement>, questionID: string) => {
+            const newQuestions = [...questions];
+            const i = questions.findIndex((q) => q.id === questionID);
+            newQuestions[i] = {
+                ...newQuestions[i],
+                text: e.target.value,
+            };
+            setQuestions(newQuestions);
+            saveSettings(newQuestions);
+        }, [questions, saveSettings]),
+        500,
+    );
+
+    const renderQuestions = useMemo(() => {
```

Comment:
Nit: `renderX` sounds like an action, therefore a function. Maybe `renderedQuestions`? 0/5

Naming is hard.

Reply:
Makes sense. Done.